apps/election-manager/src/components/ExportElectionBallotPackageModalButton.test.tsx

```diff
@@ -0,0 +1,182 @@
+
+beforeAll(() => {
+  window.kiosk = fakeKiosk()
+})
+
+afterAll(() => {
+  delete window.kiosk
+})
+
+test('Button renders properly when not clicked', async () => {
+  const { queryByText, queryByTestId } = renderInAppContext(
+    <ExportElectionBallotPackageModalButton />
+  )
+
+  expect(queryByText('Export Ballot Package')).toHaveProperty('type', 'button')
+  expect(queryByTestId('modal')).toBeNull()
+})
+
+test('Modal renders insert usb screen appropriately', async () => {
+  const usbStatuses = [
+    UsbDriveStatus.absent,
+    UsbDriveStatus.recentlyEjected,
+    UsbDriveStatus.notavailable,
+  ]
+
+  for (const usbStatus of usbStatuses) {
+    const {
+      unmount,
+      getByText,
+      queryAllByText,
+      queryAllByAltText,
+      queryAllByTestId,
+    } = renderInAppContext(<ExportElectionBallotPackageModalButton />, {
+      usbDriveStatus: usbStatus,
+    })
+    fireEvent.click(getByText('Export Ballot Package'))
+    await waitFor(() => getByText('No USB Drive Detected'))
+    expect(queryAllByAltText('Insert USB Image')).toHaveLength(1)
+    expect(queryAllByTestId('modal')).toHaveLength(1)
+    expect(
+      queryAllByText(
+        'Please insert a USB stick in order to export the ballot configuration.'
+      )
+    ).toHaveLength(1)
+
+    fireEvent.click(getByText('Cancel'))
+    expect(queryAllByTestId('modal')).toHaveLength(0)
+
+    unmount()
+  }
+})
+
+test('Modal renders export confirmation screen when usb detected and manual link works as expected', async () => {
+  const mockKiosk = fakeKiosk()
+  const fileWriter = fakeFileWriter()
+  window.kiosk = mockKiosk
+  const saveAsFunction = jest.fn().mockResolvedValue(fileWriter)
+  mockKiosk.saveAs = saveAsFunction
+  const {
+    getByText,
+    queryAllByText,
+    queryAllByAltText,
+    queryAllByTestId,
+  } = renderInAppContext(<ExportElectionBallotPackageModalButton />, {
+    usbDriveStatus: UsbDriveStatus.mounted,
+  })
+  fireEvent.click(getByText('Export Ballot Package'))
+  await waitFor(() => getByText('USB Drive Detected!'))
+  expect(queryAllByAltText('Insert USB Image')).toHaveLength(1)
+  expect(queryAllByTestId('modal')).toHaveLength(1)
+  expect(
+    queryAllByText(/Would you like to export the ballot configuration now?/)
+  ).toHaveLength(1)
+  expect(
+    queryAllByText(
+      /A zip archive will automatically be saved to the inserted USB drive./
+    )
+  ).toHaveLength(1)
+
+  expect(queryAllByTestId('manual-link')).toHaveLength(1)
+  expect(queryAllByTestId('manual-link')[0]).toHaveTextContent(
+    'manually select a location to save the archive to.'
+  )
+  fireEvent.click(queryAllByTestId('manual-link')[0])
+  await waitFor(() => getByText(/Download Complete/))
+  await waitFor(() => {
+    expect(saveAsFunction).toHaveBeenCalledTimes(1)
+  })
+
+  fireEvent.click(getByText('Cancel'))
+  expect(queryAllByTestId('modal')).toHaveLength(0)
+})
+
+test('Modal renders loading screen when usb is mounting or ejecting', async () => {
```

Comment:
```suggestion
test('Modal renders loading screen when usb drive is mounting or ejecting', async () => {
```

Reply:
Addressed in https://github.com/votingworks/vxsuite/pull/97/commits/98fd424c5290ff406ebe5302330c8bdbb381072a